packages/react/src/utils/generate-globs.ts

```diff
@@ -0,0 +1,51 @@
+import { joinPathFragments, logger } from '@nrwl/devkit';
+import { createProjectGraph } from '@nrwl/workspace/src/core/project-graph';
+import { appRootPath } from '@nrwl/workspace/src/utilities/app-root';
+import {
+  getProjectNameFromDirPath,
+  getSourceDirOfDependentProjects,
+} from '@nrwl/workspace/src/utilities/project-graph-utils';
+import { relative, resolve } from 'path';
+
+/**
+ * generates a set of glob patterns based off the source root of the app and it's dependencies
+ * @param dirPath workspace relative directory path that will be used to infer the parent project and dependencies
+ * @param fileGlobPattern pass a custom glob pattern to be used
+ */
+export function createGlobPatternsForDependencies(
+  dirPath: string,
+  fileGlobPattern: string = '/**/!(*.stories|*.spec).{tsx,jsx,js}'
+) {
+  const workspaceRoot = process.cwd();
```

Comment:
Why is this not `appRootPath`?

You can move this function to `@nrwl/workspace` and re-export it (deep import).